### PR TITLE
Write an introduction to code reviews explaining how to make the best of a code review

### PR DESCRIPTION
**Pull request by cloudappsguy**
## Description
This PR adds new content to further enhance and describe the purpose and approach to code reviews.

## PR Checklist

---

> Use the check-list below to ensure your branch is ready for PR.  If the item is not applicable, leave it blank.

- [X] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My code follows the code style of this project.
- [x] I ran the lint checks which produced no new errors nor warnings for my changes.
- [X] I have checked to ensure there aren't other open Pull Requests for the same update/change.

## Does this introduce a breaking change?

---

- [ ] Yes
- [X] No

## Any relevant logs or outputs
N/A

## Other information or known dependencies
N/A

### CODE REVIEW

File: docs/code-reviews/resources/meeting-template.txt
Comment (skyarkitekten) [inline docs/code-reviews/resources/meeting-template.txt:1]: I like the spirit here. It gives people a template to work from and sets and agenda. I fully recognize that not all teams and projects have a healthy review process.

We need to set an expectation that most reviews should happen synchronously and at smaller batches than PR level. 

No action needed at this time.

File: docs/code-reviews/README.md [hunk 1]
```diff
@@ -10,8 +10,21 @@ Code review is a way to have a conversation about the code where participants wi
 - **Learn and grow** by having others review the code, we get exposed to unfamiliar design patterns or languages among other topics, and even break some bad habits.
 - **Shared understanding** between the developers over the project's code.
 
+## When do I conduct a code review?
+
+Developers working on projects should conduct peer code reviews on every pull request (or check-in to a shared branch).
```
Comment (skyarkitekten) [inline docs/code-reviews/README.md:15]:
```suggestion
Developers working on projects should conduct peer code reviews on every pull request _at a minimum_. Teams are encouraged to review changes a few times a day.
```

File: docs/code-reviews/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Code Reviews
 
-Developers working on projects should conduct peer code reviews on every pull request (or check-in to a shared branch).
+Code reviews are an essential part of the software development process, offering numerous benefits such as improved code quality, shared knowledge, and early bug detection. However, the effectiveness of these reviews largely depends on the approach and mindset of the participants.
```
Comment (skyarkitekten) [inline docs/code-reviews/README.md:3]: I like this update. "Why we do it, how it helps you"

File: docs/code-reviews/README.md (hunk 1 above)
Comment (skyarkitekten) [reply docs/code-reviews/README.md:15]: i see this is handled in the PR docs

File: docs/design/design-patterns/rest-api-design-guidance.md
Comment (skyarkitekten) [inline docs/design/design-patterns/rest-api-design-guidance.md:1]: we should consider creating our own guidance and citing it here @mikeholdorf

**Review by skyarkitekten — APPROVED** [review]
left some comments, no pressing changings

File: docs/code-reviews/README.md (hunk 2 above)
Comment (chadcook) [reply docs/code-reviews/README.md:3]: Open to pushback, but instead of essential, we say it's required.

**Review by chadcook — DISMISSED** [review]

File: docs/code-reviews/README.md (hunk 2 above)
Comment (cloudappsguy) [reply docs/code-reviews/README.md]: updated from essential to 'required' per feedback.

**Review by skyarkitekten — APPROVED** [review]
Reviewed, comments closed, ready for merge.

**Pull request closed without merging**